front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/formatTimetableItemPayload.ts

```diff
@@ -34,18 +40,66 @@ export function formatTimetableItemPayload(
   };
 }
 
+// Format a PacedTrainWithDetails to a PacedTrain payload by keeping only the
+// necessary properties and formatting the date fields to ISO strings.
+function formatPacedTrainWithDetailsToPacedTrainPayload(
+  pacedTrainWithDetails: PacedTrainWithDetails
+): PacedTrain {
+  return {
+    category: pacedTrainWithDetails.category,
+    comfort: pacedTrainWithDetails.comfort,
+    constraint_distribution: pacedTrainWithDetails.constraint_distribution,
+    exceptions: pacedTrainWithDetails.exceptions,
+    initial_speed: pacedTrainWithDetails.initial_speed,
+    labels: pacedTrainWithDetails.labels,
+    margins: pacedTrainWithDetails.margins,
+    options: pacedTrainWithDetails.options,
+    paced: {
+      time_window: pacedTrainWithDetails.paced.timeWindow.toISOString(),
+      interval: pacedTrainWithDetails.paced.interval.toISOString(),
+    },
+    path: pacedTrainWithDetails.path,
+    power_restrictions: pacedTrainWithDetails.power_restrictions,
+    rolling_stock_name: pacedTrainWithDetails.rollingStock!.name,
+    schedule: pacedTrainWithDetails.schedule,
+    speed_limit_tag: pacedTrainWithDetails.speed_limit_tag,
+    start_time: pacedTrainWithDetails.startTime.toISOString(),
+    train_name: pacedTrainWithDetails.name,
+  };
+}
+
 export function formatPacedTrainPayload(
   osrdconf: OperationalStudiesConfState,
   // TODO TS2 : remove this when rollingStockName will replace rollingStockId in the store
-  rollingStockName: string
+  rollingStockName: string,
+  timetableItemToEditData?: TimetableItemToEditData
 ): PacedTrain {
   const baseTrain = formatTimetableItemPayload(osrdconf, rollingStockName);
-  return {
+
+  let pacedTrainToUpdate: PacedTrain = {
```

Comment:
What about newPacedTrain ?
```suggestion
  let newPacedTrain: PacedTrain = {
```

Reply:
Done